include/spdlog/sinks/mongo_sink.h

```diff
@@ -30,14 +31,18 @@ class mongo_sink : public base_sink<Mutex>
 {
 public:
     mongo_sink(const std::string &db_name, const std::string &collection_name, const std::string &uri = "mongodb://localhost:27017")
+        : mongo_sink(std::make_shared<mongocxx::instance>(), db_name, collection_name, uri) {}
+
+    mongo_sink(const std::shared_ptr<mongocxx::instance> &instance, const std::string &db_name, const std::string &collection_name, const std::string &uri = "mongodb://localhost:27017")
+        : instance_(instance)
+        , db_name_(db_name)
+        , coll_name_(collection_name)
     {
         try
         {
             client_ = spdlog::details::make_unique<mongocxx::client>(mongocxx::uri{uri});
-            db_name_ = db_name;
-            coll_name_ = collection_name;
         }
-        catch (const std::exception)
+        catch (const std::exception&)
```

Comment:
How about adding a catch clause to catch the [mongo exception](http://mongocxx.org/api/current/classmongocxx_1_1operation__exception.html) to extract its what() string and add to the spdlog_ex message as well ?

Reply:
I added the what() string to the spdlog_ex in the std::exception catch clause, which should also catch mongocxx::exception. Did you specifically want a separate catch clause for mongocxx::exception?

Also caught the mongocxx::exception that may be triggered by constructing the mongocxx::instance in the first constructor overload with a function try block, but other std::exceptions are not handled in this overload and would be re-thrown as-is. Is it better to catch all std::exceptions and re-throw them as spdlog_ex?

Reply:
yes.  but since the first ctor just forwards to the other ctor,  the try catch is redundant in the first ctor anyway

Reply:
But can't the try-catch in the first constructor catch exceptions from the `std::make_shared<mongocxx::instance>()` that appears in its initializer list while the second can't?

Reply:
Unless I'm mistaken, I'll change the first constructor's try-catch to catch any std;:exception and re-throw as spdlog_ex.